### PR TITLE
feat: execute SHER buy from fundraise screen

### PR DESCRIPTION
**Pull request by frimoldi**

### CODE REVIEW

File: src/pages/Fundraising/Fundraising.tsx [hunk 1]
```diff
@@ -157,6 +202,14 @@ export const FundraisingPage: React.FC = () => {
               <li>{`USDC Contributed: ${utils.commify(utils.formatUnits(rewards.price, 6))}`}</li>
               <li>{`SHER Reward: ${utils.commify(utils.formatUnits(rewards.sherAmount, 18))} tokens`}</li>
             </ul>
+            <AllowanceGate
+              allowance={allowance}
```
Comment (rcstanciu) [inline src/pages/Fundraising/Fundraising.tsx:206]: I know it's a PITA the way `AllowanceGate` behaves at the moment. I was looking to have it take care of getting the current allowance, but everytime I got into some sort of infinite loops in the interaction with wagmi, having `useERC20` hook being used in nested components, so I resorted to just passing the allowance as a prop.

File: src/hooks/useSherClaimContract.ts [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { useContract, useProvider } from "wagmi"
 import SherClaimInterface from "../abi/SherClaim.json"
 import { SherClaim } from "../contracts/SherClaim"
 
+export const SHER_CLAIM_ADDRESS = process.env.REACT_APP_SHER_CLAIM_ADDRESS as string
```
Comment (rcstanciu) [inline src/hooks/useSherClaimContract.ts:5]: Would be nice if you could add the newly used env variables to the `.env.example`, but other than that, this is good to go 🚀

**Review by rcstanciu — APPROVED** [review]

File: src/hooks/useSherClaimContract.ts (hunk 2 above)
Comment (frimoldi) [reply src/hooks/useSherClaimContract.ts:5]: yeah, good call

File: src/pages/Fundraising/Fundraising.tsx (hunk 1 above)
Comment (frimoldi) [reply src/pages/Fundraising/Fundraising.tsx:206]: haha yeh, there's a bit of repeated logic around allowances in both here and `Protocol.tsx` but we can refactor that component later on.

**Pull request merged**